### PR TITLE
Modify configuration and parameters from web GUI

### PR DESCRIPTION
**Pull request by renzodgc**
This addresses [Neuralet Issue #25](https://github.com/neuralet/neuralet/issues/25).
- Updates the previously added endpoint to cope with the snake_case standard.
- Adds `video_path` validation in the API by trying to process the video through cv2 before attempting to update the config.
- Adds in the frontend a `/settings` page to update video config and point towards a processor's node API.
- The `/settings` page only shows up in the navbar if the frontend is run on dev mode (with npm).

image

I kept all other config options. However, they don't have validations, so misusing the endpoint may cause an invalid config file to be saved.

### CODE REVIEW

File: api/models/config_keys.py [hunk 1]
```diff
@@ -15,7 +16,9 @@ class Config:
 
 
 class AppConfig(SnakeModel):
-    VideoPath: Optional[str] = Field(None, example='/repo/data/TownCentreXVID.avi')
+    VideoPath: Optional[str] = Field(None, example='/repo/data/gard1-4.mp4')
+    Host: Optional[str] = Field(None, example='0.0.0.0')
+    Port: Optional[str] = Field(None, example='8001')
```
Comment (mats-claassen) [inline api/models/config_keys.py:21]: I think we should leave 8000 as example
```suggestion
    Port: Optional[str] = Field(None, example='8000')
```

Comment (mats-claassen) [reply api/models/config_keys.py:21]: Actually, host and port are in ApiConfig so should be removed from here. or from ApiConfig

File: api/processor_api.py [hunk 2]
```diff
@@ -99,38 +100,20 @@ async def stop_process_video():
...
 
         @app.put("/config")
-        async def update_config(config_request: ConfigRequest):
-            save_file = config_request.save_file
-            config_request = config_request.dict(exclude_unset=True, exclude_none=True)
-            config = config_request['config']
-            
+        async def update_config(config: Config):
+            config = config.dict(exclude_unset=True, exclude_none=True)
+            print(config)
```
Comment (mats-claassen) [inline api/processor_api.py:112]: Please remove print

File: libs/processor_core.py [hunk 3]
```diff
@@ -78,6 +78,7 @@ def _serve(self):
                     self._engine.stop()
                     del self._tasks[Commands.PROCESS_VIDEO_CFG]
                     logger.info("processing stopped")
+                    self.config.reload()
```
Comment (mats-claassen) [inline libs/processor_core.py:81]: Why is this called in STOP? Wouldn't it be better to do this right before we create an EngineConfig in `if cmd_code == Commands.PROCESS_VIDEO_CFG`?

**Comment by mhejrati** [comment]
@mats-claassen have you tested this already or does it need testing on different platforms before merging?

**Comment by mats-claassen** [comment]
Has been tested on x86-Openvino and x86

**Comment by mats-claassen** [comment]
Tested also on Jetson Nano. I think this is ready to be merged

**Pull request merged**